server/app/services/export/JsonExporter.java

```diff
@@ -234,6 +228,8 @@ private static void exportApplicationEntriesToJsonApplication(
         @SuppressWarnings("unchecked")
         ImmutableList<String> list = (ImmutableList<String>) maybeJsonValue.get();
         jsonApplication.putArray(path, list);
+      } else if (instanceOfEmptyImmutableList(maybeJsonValue.get())) {
+        jsonApplication.putArray(path, ImmutableList.of());
```

Comment:
Restores the logic that was removed in #5306, which was the better way to do this.